### ISSUE TITLE
NetworkOnMainThreadException

### ISSUE THREAD

**Issue by ubuntudroid** [bugsnag/bugsnag-android#138]
We are just trying to integrate Bugsnag into our app. However, when a crash occurs (on the main thread) we are receiving this NetworkOnMainThreadException:

```
android.os.NetworkOnMainThreadException
                                                    at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1303)
                                                    at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:86)
                                                    at java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:74)
                                                    at java.net.InetAddress.getAllByName(InetAddress.java:752)
                                                    at com.android.okhttp.internal.Network$1.resolveInetAddresses(Network.java:29)
                                                    at com.android.okhttp.internal.http.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:187)
                                                    at com.android.okhttp.internal.http.RouteSelector.nextProxy(RouteSelector.java:156)
                                                    at com.android.okhttp.internal.http.RouteSelector.next(RouteSelector.java:98)
                                                    at com.android.okhttp.internal.http.HttpEngine.createNextConnection(HttpEngine.java:345)
                                                    at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:328)
                                                    at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:246)
                                                    at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:457)
                                                    at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:126)
                                                    at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getOutputStream(HttpURLConnectionImpl.java:257)
                                                    at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getOutputStream(DelegatingHttpsURLConnection.java:218)
                                                    at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getOutputStream(HttpsURLConnectionImpl.java)
                                                    at com.bugsnag.android.HttpClient.post(HttpClient.java:37)
                                                    at com.bugsnag.android.Client.deliver(Client.java:649)
                                                    at com.bugsnag.android.Client.notify(Client.java:630)
                                                    at com.bugsnag.android.Client.notify(Client.java:590)
                                                    at com.bugsnag.android.Client.notifyBlocking(Client.java:495)
                                                    at com.bugsnag.android.ExceptionHandler.uncaughtException(ExceptionHandler.java:51)
                                                    at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
                                                    at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1063)
```

There has already been an issue reporting that in 2013, but it has been resolved apparently. So I'm just reporting this as a new issue.

**Comment by Pezzah** [comment]
Could this have been introduced by the changes on https://github.com/bugsnag/bugsnag-android/pull/137?

**Comment by ubuntudroid** [comment]
Doesn't happen with 3.7.0, although no report is actually shown on the
dashboard. But I guess that's another problem because actually no
non-catched exception has been reported to the dashboard so far (also not
those which occurred on other threads than the main thread). The cause for
the latter is probably a configuration issue on our end.

**Comment by ubuntudroid** [comment]
Okay, with 3.7.0 everything works perfectly fine. :) So could very well be #137 which causes the issue with 3.7.1.

**Comment by saurabh0802chd** [comment]
any ETA on release of fix for this issue?
#136 and #138 are blockers for us.

**Comment by kattrali** [comment]
Both should be fixed by #139, which is awaiting review.
